src/internal/middleware.js

```diff
@@ -91,5 +93,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     return task
   }
 
+  sagaMiddleware.setContext = (props) => {
+    // TODO: or maybe `setContext` should be available on task and this one should act only as proxy method?
+    // would that be useful?
+    // const task = yield takeEvery(...)
+    // const task.setContext({...}) or maybe even - yield setContext(task, {...})?
```

Comment:
Yep, also I can see some parallel with `sagaMiddleware.setContext`. Maybe then we just leave `task.setContext` and remove `sagaMiddleware.setContext`, as the latter can be done with `sagaMiddleware.run(...).setContext`?